source/GameInterface/Services/ItemRosters/Patches/ItemRosterPatch.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Common.Messaging;
+using GameInterface.Services.ItemRosters.Messages.Events;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem.Roster;
+using TaleWorlds.Core;
+
+namespace GameInterface.Services.ItemRosters.Patches
+{
+    [HarmonyPatch(typeof(ItemRoster))]
+    internal class ItemRosterPatch
+    {
+        [HarmonyPatch(nameof(ItemRoster.AddToCounts), new[] { typeof(EquipmentElement), typeof(int) })]
+        [HarmonyPrefix]
+        public static void AddToCountsPrefix(ItemRoster __instance, EquipmentElement rosterElement, int number)
+        {
+            if(ItemRosterLookup.TryGetValue(__instance, out var pb))
+            {
+                if (ModInformation.IsServer)
+                {
+                    MessageBroker.Instance.Publish(__instance, new ItemRosterUpdate(
+                        pb.Id,
+                        rosterElement.Item.StringId,
+                        rosterElement.ItemModifier?.StringId,
+                        number
+                    ));
+                }
+            }
```

Comment:
Could you update "pb" to have a better variable name.

Also guard clauses are easier to read for future reference

```suggestion
           if (ModInformation.IsClient) return;
           
           if(ItemRosterLookup.TryGetValue(__instance, out var pb) == false) return;
           
           MessageBroker.Instance.Publish(__instance, new ItemRosterUpdate(
                        pb.Id,
                        rosterElement.Item.StringId,
                        rosterElement.ItemModifier?.StringId,
                        number
                    ));
```